src/wasm/kotlin/src/arcs/Entity.kt

```diff
@@ -96,16 +96,49 @@ class StringDecoder(private var str: String) {
 class StringEncoder(private val sb: StringBuilder = StringBuilder()) {
 
     companion object {
-      fun encodeDictionary(dict: Map<String, String>): String {
-        val sb = StringBuilder()
-        sb.append(dict.size).append(":")
+        fun encodeDictionary(dict: Map<String, Any>): String {
+            val sb = StringBuilder()
+            sb.append(dict.size).append(":")
+
+            for((key, value) in dict) {
```

Comment:
this could be replaced with `.joinToString`:

```kotlin
sb.append(dict.joinToString(delimeter = "") { (key, value) -> "${key.length}:$key${encodeValue(value)}" })
```

also, I believe `+=` should work with StringBuilder in kotlin, so you can avoid doing `.append`